File: csharp/create_whatsapp_campaign.cs
string prodUrl = "<REPLACE_WITH_PROD_URL>";
string token = "<REPLACE_WITH_YOUR_API_KEY>";

var client = new RestClient($"{prodUrl}/v1/campaigns/whatsapp");
client.Timeout = -1;

var request = new RestRequest(Method.POST);
request.AddHeader("Authorization", $"Bearer {token}");
request.AddHeader("Content-Type", "application/json");

var requestBody = new {
    label = "My WhatsApp Campaign",
    sender = "My Company",
    templateName = "order_confirmation",
    contacts = new[] { "2250000000000", "2250000000001", "2250000000002" },
    parameters = new[] { "John", "12345" },
    mediaUrl = "https://mydomain.com/image.jpg",
    notifyUrl = "https://mydomain.com:4444/dlr",
    skipDuplicates = true,
    sendAt = "2026-07-20T21:40:00.000Z"
};

request.AddJsonBody(requestBody);

IRestResponse response = client.Execute(request);
Console.WriteLine(response.Content);

File: java/create_whatsapp_campaign.java
import okhttp3.*;

import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        OkHttpClient client = new OkHttpClient().newBuilder().build();

        MediaType mediaType = MediaType.parse("application/json");
        RequestBody body = RequestBody.create(mediaType, "{\n" +
                "    \"label\": \"My WhatsApp Campaign\",\n" +
                "    \"sender\": \"My Company\",\n" +
                "    \"templateName\": \"order_confirmation\",\n" +
                "    \"contacts\": [\"2250000000000\", \"2250000000001\", \"2250000000002\"],\n" +
                "    \"parameters\": [\"John\", \"12345\"],\n" +
                "    \"mediaUrl\": \"https://mydomain.com/image.jpg\",\n" +
                "    \"notifyUrl\": \"https://mydomain.com:4444/dlr\",\n" +
                "    \"skipDuplicates\": true,\n" +
                "    \"sendAt\": \"2026-07-20T21:40:00.000Z\"\n" +
                "}");

        Request request = new Request.Builder()
                .url("<REPLACE_WITH_PROD_URL>/v1/campaigns/whatsapp")
                .method("POST", body)
                .addHeader("Authorization", "Bearer <REPLACE_WITH_YOUR_API_KEY>")
                .addHeader("Content-Type", "application/json")
                .build();

        Response response = client.newCall(request).execute();
        String responseBody = response.body().string();
        System.out.println(responseBody);
    }
}
